static/src/javascripts/projects/common/modules/email/email.js

```diff
@@ -0,0 +1,147 @@
+                return function (event) {
+                    var data = 'email=' + encodeURIComponent($('.' + classes.textInput, $form).val());
+
+                    require('common/modules/analytics/omniture', function (omniture) {
+                        omniture.trackLinkImmediate('rtrt | email form inline | footer | subscribe clicked');
+                    });
+
+                    event.preventDefault();
+
+                    return ajax({
+                        url: url,
+                        method: 'post',
+                        data: data,
+                        headers: {
+                            'Accept': 'application/json'
+                        }
+                    }).then(this.submissionResult(true, $form), this.submissionResult(false, $form));
+                }.bind(this);
+
+            },
+            submissionResult: function (isSuccess, $form) {
+                return function () {
+                    updateForm.replaceContent(isSuccess, $form);
+                };
+            }
+        },
+        updateForm = {
+            replaceContent: function (isSuccess, $form) {
+                var submissionMessage = {
+                        statusClass: (isSuccess) ? 'email-sub__message--success' : 'email-sub__message--failure',
+                        submissionHeadline: (isSuccess) ? 'Thank you for subscribing' : 'Something went wrong',
+                        submissionMessage: (isSuccess) ? 'We will send you our picks of the most important headlines tomorrow morning.' : 'Please try again.',
+                        submissionIcon: (isSuccess) ? svgs('tick') : svgs('crossIcon')
+                    },
+                    submissionHtml = template(successHtml, submissionMessage);
+
+                fastdom.write(function () {
+                    $form.addClass('email-sub__form--is-hidden');
+                    $form.after(submissionHtml);
+                });
+
+                if (isSuccess) {
+                    require('common/modules/analytics/omniture', function (omniture) {
+                        omniture.trackLinkImmediate('rtrt | email form inline | footer | subscribe successful');
+                    });
+                }
+            }
+        },
+        ui = {
+            freezeHeight: function ($wrapper, reset) {
+                var wrapperHeight,
+                    resetHeight = function () {
+                        $wrapper.css('min-height', '');
+                        fastdom.read(getHeight);
+                        fastdom.defer(setHeight);
+                    },
+                    getHeight = function () {
+                        wrapperHeight = $wrapper.dim().height;
+                    },
+                    setHeight = function () {
+                        $wrapper.css('min-height', wrapperHeight);
+                    };
+
+                return function () {
+                    if (reset) {
+                        fastdom.write(resetHeight);
```

Comment:
maybe better to move these fastdoms to above? not clear that they're needed here, whereas above you can clearly see what the browser is having to do?